src/planner/binder/statement/bind_create.cpp

```diff
@@ -69,51 +69,91 @@ void Binder::BindCreateViewInfo(CreateViewInfo &base) {
 	base.types = query_node.types;
 }
 
-SchemaCatalogEntry *Binder::BindCreateFunctionInfo(CreateInfo &info) {
-	auto &base = (CreateMacroInfo &)info;
+    SchemaCatalogEntry *Binder::BindCreateFunctionInfo(CreateInfo &info) {
+        auto &base = (CreateMacroInfo &)info;
 
-	if (base.function->expression->HasParameter()) {
-		throw BinderException("Parameter expressions within macro's are not supported!");
-	}
+	    /* different types */
+	    bool is_query=base.function->is_query();
+
+        if (!is_query && base.function->expression->HasParameter()) {
+            throw BinderException("Parameter expressions within macro's are not supported!");
+        }
+
+        // create macro binding in order to bind the function
+        vector<LogicalType> dummy_types;
+        vector<string> dummy_names;
+
+        // positional parameters
+        for (idx_t i = 0; i < base.function->parameters.size(); i++) {
+            auto param = (ColumnRefExpression &)*base.function->parameters[i];
+            if (param.IsQualified()) {
+                throw BinderException("Invalid parameter name '%s': must be unqualified", param.ToString());
+            }
+            dummy_types.emplace_back(LogicalType::SQLNULL);
+            dummy_names.push_back(param.GetColumnName());
+        }
+        // default parameters
+        for (auto it = base.function->default_parameters.begin(); it != base.function->default_parameters.end(); it++) {
+            auto &val = (ConstantExpression &)*it->second;
+            dummy_types.push_back(val.value.type());
+            dummy_names.push_back(it->first);
+        }
+        auto this_macro_binding = make_unique<MacroBinding>(dummy_types, dummy_names, base.name);
+        macro_binding = this_macro_binding.get();
+
+
+	    if(!is_query)
+        	ExpressionBinder::QualifyColumnNames(*this, base.function->expression);
```

Comment:
Why have this be part of a separate `if` statement?

Reply:
Was not sure which expression from query_node to pass to QualifyColumnNames